test/suite/test_compact10.py

```diff
@@ -0,0 +1,212 @@
+    # - Waits for background compaction to compact all the files and create a new full backup.
+    # - Compares the two backups.
+    def test_compact10_full_backup(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create and populate tables.
+        uris = []
+        for i in range(self.num_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            self.session.create(uri, self.create_params)
+            self.populate(uri, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of the file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Take the first full backup.
+        os.mkdir(self.backup_dir_1)
+        self.take_full_backup(self.backup_dir_1)
+
+        # Enable background compaction.
+        compact_config = f'background=true,free_space_target=1MB'
+        self.turn_on_bg_compact(compact_config)
+
+        # Wait for all tables to be compacted but the HS.
+        while self.get_files_compacted(uris) < self.num_tables:
+            time.sleep(0.5)
+
+        assert self.get_files_compacted(uris) == self.num_tables
+
+        # Take a second full backup.
+        os.mkdir(self.backup_dir_2)
+        self.take_full_backup(self.backup_dir_2)
+
+        # Compare the backups.
+        for uri in uris:
+            self.compare_backups(uri, self.backup_dir_1, self.backup_dir_2)
+
+        # Background compaction may be have been inspecting a table when disabled which is
+        # considered as an interruption, ignore that message.
+        self.ignoreStdoutPatternIfExists('background compact interrupted by application')
+
+    # This test:
+    # - Creates two full backups before background compaction is enabled, one that will be used for
+    # comparison and another one as a base for incremental backups.
+    # - Performs incremental backups as compaction is rewriting the files until all files have been
+    # compacted.
+    # - Compares the two backups.
+    def test_compact10_incr_backup(self):
+
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create and populate tables.
+        uris = []
+        for i in range(self.num_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            self.session.create(uri, self.create_params)
+            self.populate(uri, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of the file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
```

Comment:
This part of the test is the same in both cases. How about creating another function that can be used by these two tests?